prog/test/vm.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+# st3 = Strand.create_with_id(prog: 'Test::Vm', label: 'start', stack: [{subject_id: Vm.first.id}])
+class Prog::Test::Vm < Prog::Base
+  subject_is :sshable, :vm
+
+  label def start
+    hop_verify_dd
+  end
+
+  label def verify_dd
+    # Verifies basic block device health
+    # See https://github.com/ubicloud/ubicloud/issues/276
+    sshable.cmd("dd if=/dev/random of=~/1.txt bs=512 count=1000000")
+    sshable.cmd("sync ~/1.txt")
+    size_info = sshable.cmd("ls -s ~/1.txt").split
+
+    unless size_info[0].to_i.between?(500000, 500100)
+      fail "unexpected size after dd"
+    end
+
+    hop_install_packages
+  end
+
+  label def install_packages
+    sshable.cmd("sudo apt update")
+    sshable.cmd("sudo apt install -y build-essential")
+
+    hop_compile_program
+  end
+
+  label def compile_program
+    sshable.cmd <<WRITE_PROGRAM
+        cat <<EOL > ~/1.c
+        #include <stdio.h>
+        int main() {
+        printf("Hello World.\\n");
+        return 0;
+        }
+EOL
+WRITE_PROGRAM
+
+    sshable.cmd("gcc ~/1.c -o ~/1.out")
+    sshable.cmd("~/1.out").strip
+
+    unless sshable.cmd("~/1.out").strip == "Hello World."
+      fail "Unexpected output for hello world program"
+    end
+
+    hop_ping_google
+  end
```

Comment:
What is this test trying to exercise? Because, given the apt install step required as well, it takes a while...and if it can load `dd` (and create disk traffic) and run `ping`, what's so special about loading the compiler?

Reply:
Mostly inspired by what I would do personally if I use a VM. I'll remove it.